W-21162798 Combine overviews for agent network/agent fabric (#578)
mulesoft/docs-code-builder: modules/ROOT/pages/af-agent-networks.adoc

= Building Agent Networks for Agent Fabric

An agent network is a coordinated group of agents, brokers, LLMs, and MCP servers that acts as a central hub for defining, validating, and executing agentic processes across your enterprise.

> While MuleSoft Agent Fabric enables you to maximize the potential of every AI agent with centralized discovery, orchestration across agents and tools, cross-ecosystem governance, and full transparency into agentic interactions, an agent network provides the build blocks of your agentic deployments.

Reviewer: ```suggestion
An agent network provides the building blocks of your agentic deployment while MuleSoft Agent Fabric helps you maximize the potential of every AI agent with centralized discovery, orchestration across agents and tools, cross-ecosystem governance, and full transparency into agentic interactions. 
```

Reviewer: I only switched it around since this is the agent network overview but this is a totally optional suggestion obvs.

> You define your agent network in a simple, human-readable YAML file in Anypoint Code Builder. This approach abstracts away the underlying technical complexities, allowing you to focus on the business constraints and context of your process without needing to understand the inner workings of the orchestration engine. At the start of a new agent network project, we provide a YAML template to give you a head start. You can even use MuleSoft Dev Agent to configure your network, publish the assets to Anypoint Exchange, and deploy your agent network instance.

Reviewer: I would make this two paragraphs. Second one starting with the "At the start..."

[[key-benefits]]
== Key Benefits of Agent Networks

Create robust, automated agent networks with these key benefits.

* Simplified Development
+
Define your agent processes in an intentional way using a simple YAML file, which is easier and faster than writing complex code.
* Safety and Governance
+
Ensure control and compliance across every agent and tool interaction with enterprise-grade governance. 
* Observability
+
Gain insight into agent actions with a visual trace of their decision-making. 
* Flexibility
+
Coordinate any type of agent and tool, regardless of where it’s built and deployed so you can build a diverse network of agents.
* Tool and Agent Invocation
+
Manage the invocation of both deterministic (tool-based) and probabilistic (LLM-based) actions.

[[example-use-case]]
== Example Use Case: Employee Onboarding

In this example, the goal is to seamlessly activate a new hire across HR, IT, Security, and Facilities with a single, policy‑governed experience.

Here’s how it works:

* Discover
+

> Publish existing agents and tools (HRIS, Identity, ITSM, Payroll, Facilities, LMS) in Exchange for reuse.

Reviewer: If this is the first mention of Exchange, we have to do the Anypoint Exchange (Exchange) thing. Especially since Agentforce consumption of content is a consideration now.

The asset might also be published to Exchange.
. An asset that has been added to this agent network project from Exchange.

[[agent-network-architecture]]
== Agent Network Architecture

> The following diagram shows an agent network with agents and MCP servers, traffic through Flex Gateway, governed by API Manager, and observed in Monitoring and Visualizer.

Reviewer: ```suggestion
This diagram shows an agent network with agents and MCP servers, traffic through Flex Gateway, governed by API Manager, and observed in Anypoint Monitoring and Agent Visualizer.
```

> image::agent-fabric-architecture.png[Agent Fabric showing agents and MCP servers defined in YAML and published to Exchange, with traffic through Flex Gateway, governed by API Manager, and observed in Monitoring and Visualizer]

Reviewer: ```suggestion
image::agent-fabric-architecture.png[Agent Fabric showing agents and MCP servers defined in YAML and published to Exchange, with traffic through Flex Gateway, governed by API Manager, and observed in Anypoint Monitoring and Agent Visualizer]
```

Reviewer: Since we have two Visualizers now, I never use it by itself.
